node/src/types/appendable_block.rs

```diff
@@ -190,7 +190,12 @@ impl AppendableBlock {
             - self.transfers.len()
             + self.deploy_config.block_max_deploy_count as usize
             - self.deploys.len();
-        additional_approvals > remaining_approval_slots - remaining_deploy_slots + 1
+        if let Some(total_approvals) = additional_approvals.checked_add(remaining_deploy_slots) {
```

Comment:
I think calling it `total_approvals` might be confusing, since this is a sum of a number of actual approvals and a number representing deploy slots... I'd call it something like `approvals_plus_deploy_slots` instead.

I don't really understand why we need this change, though - we should never hit the case of `remaining_deploy_slots` being greater than `remaining_approval_slots` precisely because of this check.

Reply:
After a discussion with @fizyk20 we agreed that it's better to actually add a validation of the deploy config parameters and revert to the previous implementation. See https://github.com/casper-network/casper-node/pull/3544/commits/bb3b3394119146f7937711109d237ff6e018fc31 and https://github.com/casper-network/casper-node/pull/3544/commits/fc3597e3db831f57ef80d08c925797542c9f70a7